synthesizer/program/src/parse.rs

```diff
@@ -106,10 +106,16 @@ impl<N: Network, Instruction: InstructionTrait<N>, Command: CommandTrait<N>> Fro
 
     /// Returns a program from a string literal.
     fn from_str(string: &str) -> Result<Self> {
+        // Ensure the raw program string is less than MAX_PROGRAM_CHAR_SIZE.
+        ensure!(string.len() <= N::MAX_PROGRAM_CHAR_SIZE, "Program exceeds N::MAX_PROGRAM_CHAR_SIZE.");
```

Comment:
`string.len()` is equivalent to `string.to_bytes_le().len()`

Why are we checking two different values at 3.5MB and 7MB when it's the same value?

Reply:
They are not equivalent, the original input string is usually longer, the byte vector is a more succinct representation. If you want a quick example print the values in this test: https://github.com/AleoHQ/snarkVM/blob/limit_program_size/synthesizer/program/src/bytes.rs#L164

However, I'm happy to re-use MAX_PROGRAM_BYTE_SIZE here. Let me know what you prefer.

Reply:
I see, you're comparing `program.to_bytes_le().len()` instead of `string.to_bytes_le().len()`.

The `program.to_bytes_le()` does not grow linearly at a factor of 2x consistently for all program types. It should be sublinear (by a wide margin) from the string representation.

Reply:
I may recommend checking the string length for better intuition. It's easy for developers to enforce a program size elsewhere if they have a uniform metric to measure against.

Reply:
I like: 9e0f25160